apierrors/errors.go

```diff
@@ -27,13 +30,18 @@ var (
 	ErrResourcePublished                 = errors.New("unable to update resource as it has been published")
 	ErrResourceState                     = errors.New("incorrect resource state")
 	ErrTooManyWildcards                  = errors.New("only one wildcard (*) is allowed as a value in selected query parameters")
-	ErrUnableToReadMessage               = errors.New("failed to read message body")
 	ErrUnableToParseJSON                 = errors.New("failed to parse json body")
+	ErrUnableToReadMessage               = errors.New("failed to read message body")
 	ErrUnauthorised                      = errors.New("unauthorised access to API")
-	ErrVersionBadRequest                 = errors.New("failed to parse json body")
 	ErrVersionMissingState               = errors.New("missing state from version")
 	ErrVersionNotFound                   = errors.New("version not found")
 
+	ErrExpectedResourceStateOfCreated          = errors.New("unable to update resource, expected resource to have a state of created")
```

Comment:
This would be better if you had one error struct with a status field on it

Reply:
True but the fact that these errors need to be matched on in forbidden map on line 64 makes this a non-trivial task. I will leave it for now.